hazelcast/src/main/java/com/hazelcast/client/impl/ClientEndpointImpl.java

```diff
@@ -89,7 +84,15 @@ public String getUuid() {
 
     @Override
     public boolean isAlive() {
```

Comment:
So a `ClientEndpoint` is alive as long as its connection or another connection from this member to the same client UUID is alive? Isn't it risky that one might do something like:
```
if (endpoint.isAlive()) {
  endpoint.getConnection().write(frame);
}
```
and fail as `endpoint.getConnection().isAlive()` may be `false` even though `endpoint.isAlive()==true`?